sound/soc/sof/ops.h

```diff
@@ -357,7 +357,7 @@ static inline u64 snd_sof_dsp_read64(struct snd_sof_dev *sdev, u32 bar,
 }
 
 static inline void snd_sof_dsp_update8(struct snd_sof_dev *sdev, u32 bar,
-				       u32 offset, u8 value, u8 mask)
+				       u32 offset, u8 mask, u8 value)
```

Comment:
Wow, how we did not noticed this???
git grep -A1 snd_sof_dsp_update8
```
sound/soc/sof/intel/hda-dsp.c:  snd_sof_dsp_update8(sdev, HDA_DSP_HDA_BAR, chip->d0i3_offset,
sound/soc/sof/intel/hda-dsp.c-                      SOF_HDA_VS_D0I3C_I3, value);
--
sound/soc/sof/intel/hda-loader.c:       snd_sof_dsp_update8(sdev, HDA_DSP_HDA_BAR, HDA_VS_INTEL_LTRP,
sound/soc/sof/intel/hda-loader.c-                           HDA_VS_INTEL_LTRP_GB_MASK, original_gb);
--
sound/soc/sof/intel/hda-stream.c:       snd_sof_dsp_update8(sdev, HDA_DSP_HDA_BAR, HDA_VS_INTEL_LTRP,
sound/soc/sof/intel/hda-stream.c-                           HDA_VS_INTEL_LTRP_GB_MASK, HDA_LTRP_GB_VALUE_US);
--
sound/soc/sof/ops.h:static inline void snd_sof_dsp_update8(struct snd_sof_dev *sdev, u32 bar,
sound/soc/sof/ops.h-                                   u32 offset, u8 value, u8 mask)
```
None of these should be working correctly.

Reply:
I don't why. First I though we can't set register with byte function and need 32bits one, later I realized that the order is incorrect.

Reply:
@RanderWang @ujfalusi do we need to add fixes ? This seems like an important one to fix in stable no?

Reply:
Wow, now that's a great find. This is actually not super-old, but was added in a cleanup patch. The LTRP values are probably close enough that the tests have passed. And hw can potentially handle the DxIy transition despite the wrong setting.

Anyways, excellent find. @RanderWang please add a "Fixes" tag, so this is backported to stable kernels. It would seem it is this:
```
Fixes: 847fd278610d ("ASoC: SOF: Intel: hda-stream: use snd_sof_dsp_updateb() helper")
```

Reply:
Ok this is very recent. This is not in 6.1.y stable tree and not in any of the other stable tree. So while this is in current Linux main, it's not in any released Linux version.

Reply:
wait is that fixes correct? This patch was modified to rename updateb to update8 isnt it?

Reply:
the function snd_sof_dsp_updateb is right but the problem is using it in incorrect way. we should change the mask & value order

Reply:
no what I meant was if the tag is correct

Reply:
The PR in tag used incorrect parameters and made a error, so fix it ?

Reply:
First I had two choices to fix issue: (1) change snd_sof_dsp_updateb reference (2) change snd_sof_dsp_updateb.  I chose (2) but it fixes issues caused by reference.

Reply:
@RanderWang I think the Fixed tag should be 43449b4d837c ("fixup! ASoC: SOF: fix compilation issue with readb/writeb helpers"). 
The issue was introduced by
68863a7353ae ASoC: SOF: ops: add snd_sof_dsp_updateb() helper
d3b19b1fcd66 ASoC: SOF: Intel: hda-dsp: use SOF helpers for consistency

But this commit can only apply after 43449b4d837c fixup! ASoC: SOF: fix compilation issue with readb/writeb helpers.

Reply:
o,  last fixup is not merged so let's fixup fixup

Reply:
But then `f8fbf0dc702b ("ASoC: SOF: fix compilation issue with readb/writeb helpers")` modifies the surrounding, let me check

Reply:
Both is in 6.2-rc1, so `c28a36b012f1` is good as Fixup, the other is around.

Reply:
@RanderWang, can you change the Fixup tag?

Reply:
@ujfalusi.  Ok